kvserver: deflake TestProtectedTimestamps

arulajmani opened the pull request:
Previously, we weren't waiting for the PTS record to make its way to KV in this test. This meant expecting it to provide protection (which the test asserted) was racy with spanconfig reconciliation. Span config reconciliation is built over rangefeeds, so when we bumped the default value of `kv.rangefeed.closed_timestamp_refresh_interval` in #108667, it meant span config reconciliation would take longer. This caused the GC queue to run before the replica was aware of the PTS record, which violated a test assertion.

This patch fixes the issue by only enqueueing the range in the GC queue once the PTS record is known to have made it to KV.

Closes #109244

Release note: None

arulajmani committed:
kvserver: deflake TestProtectedTimestamps

Previously, we weren't waiting for the PTS record to make its way to KV
in this test. This meant expecting it to provide protection (which the
test asserted) was racy with spanconfig reconciliation. Span config
reconciliation is built over rangefeeds, so when we bumped the default
value of `kv.rangefeed.closed_timestamp_refresh_interval` in #108667,
it meant span config reconciliation would take longer. This caused the
GC queue to run before the replica was aware of the PTS record, which
violated a test assertion.

This patch fixes the issue by only enqueueing the range in the GC queue
once the PTS record is known to have made it to KV.

Closes #109244

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110072)
<!-- Reviewable:end -->

irfansharif commented on pkg/kv/kvserver/client_protectedts_test.go:
```diff
@@ -180,6 +180,15 @@ ORDER BY raw_start_key ASC LIMIT 1`)
+	ptsReader := tc.GetFirstStoreFromServer(t, 0).GetStoreConfig().ProtectedTimestampReader
+	require.NoError(
+		t,
+		ptutil.TestingVerifyProtectionTimestampExistsOnSpans(
```
Could you rename this helper to something that captures that there's a SucceedsSoon block internally? WaitUntil...? There's one usage that wraps this helper around a SucceedsSoon block as well, so it's confused at least one other person.

arulajmani committed:
kvserver: rename TestingVerifyProtectionTimestampExistsOnSpans

TestingWaitForProtectedTimestampToExistOnSpans captures the fact
that there's a SucceedsSoon in there better. While here, we also clean
up a few usages and no longer return a useless error.

Epic: none

Release note: None

arulajmani reviewed (commented):
TFTR! 

bors r=irfansharif

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110072#-:-NdblxkP-ZJD1yn1ZGnG:b75nfk4)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @irfansharif and @pavelkalinnikov)

___
*[`pkg/kv/kvserver/client_protectedts_test.go` line 188 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/110072#gh-1316519941:-NdbllY0AKUhWYRpTTeQ:bl3br3n) ([raw file](https://github.com/cockroachdb/cockroach/blob/cbec80a3f1e4aba253339451fe01d7c652f86c0b/pkg/kv/kvserver/client_protectedts_test.go#L188)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

Could you rename this helper to something that captures that there's a SucceedsSoon block internally? WaitUntil...? There's one usage that wraps this helper around a SucceedsSoon block as well, so it's confused at least one other person.
</blockquote></details>

Good call, tacked on a commit.


<!-- Sent from Reviewable.io -->
